misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperExecutorUtil.java

```diff
@@ -0,0 +1,424 @@
+package org.ballerinalang.datamapper;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.reflect.TypeToken;
+import io.ballerina.compiler.api.symbols.IntersectionTypeSymbol;
+import io.ballerina.compiler.api.symbols.RecordFieldSymbol;
+import io.ballerina.compiler.api.symbols.RecordTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeDescKind;
+import io.ballerina.compiler.api.symbols.TypeReferenceTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.tools.text.TextDocument;
+import org.ballerinalang.datamapper.utils.HttpClientRequest;
+import org.ballerinalang.datamapper.utils.HttpResponse;
+import org.ballerinalang.langserver.common.utils.CommonUtil;
+import org.ballerinalang.langserver.commons.ExecuteCommandContext;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static org.ballerinalang.datamapper.utils.DefaultValueGenerator.generateDefaultValues;
+
+/**
+ * Utils related to the command executor of data mapper code action.
+ */
+public class AIDataMapperExecutorUtil {
+
+    private static final int HTTP_200_OK = 200;
+    private static final int HTTP_422_UN_PROCESSABLE_ENTITY = 422;
+    private static final int HTTP_500_INTERNAL_SERVER_ERROR = 500;
+    private static final int MAXIMUM_CACHE_SIZE = 100;
+    private Cache<Integer, String> mappingCache =
+            CacheBuilder.newBuilder().maximumSize(MAXIMUM_CACHE_SIZE).build();
+    private HashMap<String, Map<String, RecordFieldSymbol>> spreadFieldMap = new HashMap<>();
+    private HashMap<String, String> isOptionalMap = new HashMap<>();
+    private HashMap<String, String> leftFieldMap = new HashMap<>();
+    private HashMap<String, String> responseFieldMap = new HashMap<>();
+    private HashMap<String, String> restFieldMap = new HashMap<>();
+    private HashMap<String, String> spreadFieldResponseMap = new HashMap<>();
+    private ArrayList<String> leftReadOnlyFields = new ArrayList<>();
+    private ArrayList<String> rightSpecificFieldList = new ArrayList<>();
+    private ArrayList<String> optionalRightRecordFields = new ArrayList<>();
```

Comment:
Please revisit and check whether defining these as class variable correct? It seems most of them can be defined as method variables.

Reply:
Modified the code according to the comments.